python/pyarrow/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -2515,6 +2515,8 @@ Status ConvertChunkedArrayToPandas(const PandasOptions& options,
                                    std::shared_ptr<ChunkedArray> arr, PyObject* py_ref,
                                    PyObject** out) {
   if (options.decode_dictionaries && arr->type()->id() == Type::DICTIONARY) {
+    // XXX we should return an error as below if options.zero_copy_only
+    // is true, but that would break compatibility with existing tests.
```

Comment:
At the moment we essentially just ignore the `zero_copy_only` keyword in case of a DictionaryArray?

Reply:
Yes.